## Digital Marketing Strategy

In this Python file we investigate which factors drive consumers engagement with brand post popularity with U.S. theme parc posts on Facebook

In [390]:
# Import libraries
import numpy as np
import pandas as pd
import os

In [391]:
# Check current path
current_directory = os.getcwd()
current_directory

'c:\\Users\\yunus\\Desktop\\MSc\\Digital Driven Business\\Subjects\\Digital Marketing Strategy\\Dataset\\Digital-Marketing-Strategy2'

In [392]:
# Define path
os.chdir("C:\\Users\\yunus\\Desktop\\MSc\\Digital Driven Business\\Subjects\\Digital Marketing Strategy\\Dataset\\Digital-Marketing-Strategy2")

In [393]:
# read the csv
# Assuming semicolon is the correct delimiter for each CSV file
themepark_1 = pd.read_csv("Themeparcs I.csv", sep=';', on_bad_lines='skip')
themepark_2 = pd.read_csv("Themeparcs II.csv", sep=';', on_bad_lines='skip')
themepark_3 = pd.read_csv("Themeparcs III.csv", sep=';', on_bad_lines='skip')
themepark_4 = pd.read_csv("Themeparcs IV.csv", sep=';', on_bad_lines='skip')

## 1. Exploratoy Data Analysis (EDA) & Data Cleaning

1.1 Check shape of `themepark_1`, `themepark_2`, `themepark_3` & `themepark_4`

In [394]:
themepark_1.shape

(5000, 18)

In [395]:
themepark_2.shape

(5000, 18)

In [396]:
themepark_3.shape

(5000, 18)

In [397]:
themepark_4.shape

(443, 18)

In [398]:
# Amount of total rows for each df
5000+5000+5000+443

15443

1.2 Check features of `themepark_1`, `themepark_2`, `themepark_3` & `themepark_4`

In [399]:
themepark_1.dtypes

Date                             object
Profile                          object
Message                          object
Number of Reactions             float64
Number of Shares                float64
Number of Comments (primary)    float64
Number of Shares.1              float64
Comments per post               float64
Comments on posts by fans       float64
Post interaction rate            object
Engagement                       object
Number of Likes                 float64
Link-posts                      float64
Picture Posts                   float64
Video-posts                     float64
Profile-ID                       object
Post-ID                          object
Link                             object
dtype: object

In [400]:
themepark_2.dtypes

Date                             object
Profile                          object
Message                          object
Number of Reactions             float64
Number of Shares                float64
Number of Comments (primary)    float64
Number of Shares.1              float64
Comments per post               float64
Comments on posts by fans       float64
Post interaction rate            object
Engagement                       object
Number of Likes                 float64
Link-posts                      float64
Picture Posts                   float64
Video-posts                     float64
Profile-ID                       object
Post-ID                          object
Link                             object
dtype: object

In [401]:
themepark_3.dtypes

Date                             object
Profile                          object
Message                          object
Number of Reactions             float64
Number of Shares                float64
Number of Comments (primary)    float64
Number of Shares.1              float64
Comments per post               float64
Comments on posts by fans       float64
Post interaction rate            object
Engagement                       object
Number of Likes                 float64
Link-posts                      float64
Picture Posts                   float64
Video-posts                     float64
Profile-ID                       object
Post-ID                          object
Link                             object
dtype: object

In [402]:
themepark_4.dtypes

Date                             object
Profile                          object
Message                          object
Number of Reactions             float64
Number of Shares                float64
Number of Comments (primary)    float64
Number of Shares.1              float64
Comments per post               float64
Comments on posts by fans       float64
Post interaction rate            object
Engagement                       object
Number of Likes                 float64
Link-posts                      float64
Picture Posts                   float64
Video-posts                     float64
Profile-ID                       object
Post-ID                          object
Link                             object
dtype: object

1.3 Merge `themepark_1`, `themepark_2`, `themepark_3` & `themepark_4` since each DataFrame consists of the same features

In [403]:
# Show the first couple of rows of the first csv file 
# themepark_1.head()

In [404]:
# Merge the DataFrames
merged_themepark = pd.concat([themepark_1, themepark_2, themepark_3, themepark_4], ignore_index=True)

# If you want to save the merged DataFrame to a new CSV file
# merged_themepark.to_csv("Merged_Themeparks.csv", index=False)

In [405]:
# merged_themepark.to_csv

In [406]:
# Split the single column into multiple columns using the semicolon delimiter
# split_data = merged_themepark[merged_themepark.columns[0]].str.split(';', expand=True)

In [407]:
# Assign the new DataFrame columns to the correct names
# Replace these names with the actual column names you expect based on your data
# column_names = ['Date', 'Profile', 'Message', 'Number of Reactions', 'Number of Shares', 
#                'Number of Comments (primary)', 'Number of Shares', 'Comments per post', 
#                'Comments on posts by fans', 'Post interaction rate', 'Engagement', 
#                'Number of Likes', 'Link-posts', 'Picture Posts', 'Video-posts', 
#                'Profile-ID', 'Post-ID', 'Link']

In [408]:
# Ensure the number of column names matches the number of columns in 'split_data'
# if len(column_names) == split_data.shape[1]:
#    split_data.columns = column_names
# else:
#    print("The number of provided column names does not match the number of columns in the data.")  

In [410]:
merged_themepark.describe()

Number of Reactions  Number of Shares  Number of Comments (primary)  \
count         15443.000000               0.0                  15443.000000   
mean            494.282911               NaN                     34.199702   
std            3717.067171               NaN                    182.409237   
min               0.000000               NaN                      0.000000   
25%              22.000000               NaN                      1.000000   
50%              67.000000               NaN                      6.000000   
75%             217.000000               NaN                     18.000000   
max          210756.000000               NaN                   8379.000000   

       Number of Shares.1  Comments per post  Comments on posts by fans  \
count        15443.000000       15443.000000               15443.000000   
mean            62.078288          52.954737                  52.954737   
std            507.937148         265.913045                 265.913045   
min              0.000000           0.000000                   0.000000   
25%              2.000000           1.000000                   1.000000   
50%              6.000000           8.000000                   8.000000   
75%             21.000000          27.000000                  27.000000   
max          30762.000000       11170.000000               11170.000000   

       Number of Likes  Link-posts  Picture Posts  Video-posts  
count     15443.000000       555.0        11562.0       2479.0  
mean        355.763453         1.0            1.0          1.0  
std        2509.935344         0.0            0.0          0.0  
min           0.000000         1.0            1.0          1.0  
25%          19.000000         1.0            1.0          1.0  
50%          55.000000         1.0            1.0          1.0  
75%         177.000000         1.0            1.0          1.0  
max      124591.000000         1.0            1.0          1.0


- Number of Reactions: A count of all reactions (likes, loves, etc.) to the posts.
- Number of Shares: The count of times the posts have been shared. This column seems to have an issue as it shows all zeros or may missing data.
- Number of Comments (primary): The total number of primary comments made on the posts.
- Number of Shares.1: Appears to be a duplicate or corrected column for the number of shares with actual non-zero data.
- Comments per post: An average of how many comments were made per post.
- Comments on posts by fans: The total number of comments made by fans on the posts.
- Number of Likes: A count of likes specifically, as opposed to all reactions.
- Link-posts: The number of posts that contain links.
- Picture Posts: The number of posts that contain pictures.
- Video-posts: The number of posts that contain videos.

In [422]:
merged_themepark.shape

(15443, 18)

In [411]:
# Print the first rows 
merged_themepark.head()

Date                     Profile  \
0  12/31/23 9:40 PM  Wild Adventures Theme Park   
1  12/31/23 8:01 PM            Fun Spot America   
2  12/31/23 7:30 PM            Fun Spot America   
3  12/31/23 7:00 PM            Disneyland Paris   
4  12/31/23 7:00 PM       Silverwood Theme Park   

                                             Message  Number of Reactions  \
0  🎟️🚨 Today's the FINAL day to use your 2023 Sea...                 16.0   
1  See you next year! 😉                Don’t miss...                 38.0   
2  Don't miss our HUGE firework display ! 🥳\n\n#f...                 23.0   
3     Fijne jaarwisseling! 🥂🎉\nEn tot volgend jaar 😉                646.0   
4  It's the last day of our New Year's Sale! Save...                 17.0   

   Number of Shares  Number of Comments (primary)  Number of Shares.1  \
0               NaN                           2.0                 2.0   
1               NaN                           1.0                 2.0   
2               NaN                           2.0                 3.0   
3               NaN                          52.0                12.0   
4               NaN                           2.0                 0.0   

   Comments per post  Comments on posts by fans Post interaction rate  \
0                2.0                        2.0              1,09E+12   
1                1.0                        1.0              4,17E+11   
2                2.0                        2.0              2,85E+11   
3               56.0                       56.0              1,36E+11   
4                2.0                        2.0              4,96E+11   

  Engagement  Number of Likes  Link-posts  Picture Posts  Video-posts  \
0   1,09E+12             13.0         NaN            1.0          NaN   
1   4,17E+11             33.0         NaN            NaN          NaN   
2   2,85E+11             21.0         NaN            NaN          1.0   
3   1,36E+11            478.0         NaN            1.0          NaN   
4   4,96E+11             16.0         NaN            NaN          1.0   

    Profile-ID                          Post-ID  \
0  84679583132      84679583132_770120045161180   
1   1,8402E+11     184020466977_767286052104076   
2   1,8402E+11     184020466977_767267948772553   
3  6,52389E+14  652389081615295_775974521237956   
4  7,62702E+14  762701550509617_678920137770716   

                                                Link  
0  https://www.facebook.com/617429960430190/posts...  
1  https://www.facebook.com/653224623510220/posts...  
2  https://www.facebook.com/653224623510220/posts...  
3  https://www.facebook.com/724434559725286/posts...  
4  https://www.facebook.com/562559642740100/posts...

- Date: The date and time when the post was made
- Profile: The name of the page or profile that made the post
- Message: The content or message of the post
- Number of Reactions: The total count of reactions (like, love, haha, etc.) received by the post
- Number of Shares: The total number of times the post has been shared
- Number of Comments (primary): The initial number of comments on the post
- Comments per post: The average number of comments per post
- Comments on posts by fans: The number of comments made by fans on the posts
- Post interaction: Interaction count for the post, which might include reactions, comments, and shares
- Engagement rate: A metric that shows the level of engagement users have with the post
- Number of Likes: The number of 'like' reactions to the post
- Link-posts: If the post includes a link, this might indicate the presence of it
- Picture Posts: Indicates if the post contains a picture
- Video-posts: Indicates if the post includes a video
- Profile-ID: A unique identifier for the profile or page that made the post
- Post-ID: A unique identifier for the post itself
- Link: A URL to the post on the platform it was published


**Conclusion**
- `Date` consists of data and time in one feature
- Seeing that `Profile` contains of theme parcs outside the U.S.
- `Message` varies in length, use of symbols and emoticons
- `Number of shares` consists completely out non values (NaN). However, `Number of shares` consists of values. Therefore, we'll drop `Number of shares`
- `Post interaction rate` and `Engagement rate` both needs to by scaled for further cleaning
- Since `Profile-ID` is a unique identifier for the profile- and post, we'll put this feature most left of the DataFrame
- `Post-ID` is being next to `Profile-ID`


In [412]:
# Exact date format to parse it more efficiently by inferring datetime format
merged_themepark['Date'] = pd.to_datetime(merged_themepark['Date'], infer_datetime_format=True)

C:\Users\yunus\AppData\Local\Temp\ipykernel_4464\1775473985.py:2: UserWarning: The argument 'infer_datetime_format' is deprecated and will be removed in a future version. A strict version of it is now the default, see https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. You can safely remove this argument.
  merged_themepark['Date'] = pd.to_datetime(merged_themepark['Date'], infer_datetime_format=True)
C:\Users\yunus\AppData\Local\Temp\ipykernel_4464\1775473985.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  merged_themepark['Date'] = pd.to_datetime(merged_themepark['Date'], infer_datetime_format=True)


In [413]:
# Create a Time feature
merged_themepark['New_Date'] = merged_themepark['Date'].dt.date
merged_themepark['Time'] = merged_themepark['Date'].dt.time

# Now drop the original 'Date' column since it has been split
merged_themepark = merged_themepark.drop('Date', axis=1)

In [414]:
# Change feature format
merged_themepark = merged_themepark[['Profile-ID', 'Post-ID', 'Profile', 'Message', 'New_Date', 'Time', 'Number of Reactions', 'Number of Comments (primary)', 'Number of Shares.1', 'Comments per post', 'Comments on posts by fans', 'Post interaction rate', 'Engagement', 'Number of Likes', 'Link-posts', 'Picture Posts', 'Video-posts', 'Link']]

In [415]:
# Check for missing values
merged_themepark.isnull().sum()

Profile-ID                          0
Post-ID                             0
Profile                             0
Message                             0
New_Date                            0
Time                                0
Number of Reactions                 0
Number of Comments (primary)        0
Number of Shares.1                  0
Comments per post                   0
Comments on posts by fans           0
Post interaction rate               0
Engagement                          0
Number of Likes                     0
Link-posts                      14888
Picture Posts                    3881
Video-posts                     12964
Link                                0
dtype: int64

In [416]:
# Check for dtypes of each feature
merged_themepark.dtypes

Profile-ID                       object
Post-ID                          object
Profile                          object
Message                          object
New_Date                         object
Time                             object
Number of Reactions             float64
Number of Comments (primary)    float64
Number of Shares.1              float64
Comments per post               float64
Comments on posts by fans       float64
Post interaction rate            object
Engagement                       object
Number of Likes                 float64
Link-posts                      float64
Picture Posts                   float64
Video-posts                     float64
Link                             object
dtype: object

In [421]:
# Counting the number of messages for each profile
# messages_count = merged_themepark.groupby('Profile')['Message'].count().reset_index(name='Message Count')

# Sorting the result in descending order of DataFram message_count
# messages_count = messages_count.sort_values('Message Count', ascending=False)
# messages_count.head()

Profile  Message Count
58  Wild Adventures Theme Park            955
21     Happy Hollow Park & Zoo            627
11                 Cedar Point            478
54    Walt Disney Studios (US)            450
31                    Playland            446

In [ ]:
# merged_themepark_count = merged_themepark['Profile'].value_counts()
# merged_themepark_count

Profile
Wild Adventures Theme Park                     955
Happy Hollow Park & Zoo                        627
Cedar Point                                    478
Walt Disney Studios (US)                       450
Playland                                       446
Walt Disney Animation Studios                  432
Fun Spot America                               414
Belmont Park                                   408
Knott's Berry Farm                             402
Six Flags Discovery Kingdom                    401
Busch Gardens Tampa Bay                        399
Kings Island                                   396
Busch Gardens Williamsburg                     396
SeaWorld Orlando                               395
SeaWorld San Diego                             388
Hersheypark                                    383
SkyPark at Santa's Village                     382
Six Flags Great Adventure                      368
Santa Cruz Beach Boardwalk                     351
Lake Compounce         

In [ ]:
# Show the final DataFrame
merged_themepark.head(1)

Profile-ID                      Post-ID                     Profile  \
0  84679583132  84679583132_770120045161180  Wild Adventures Theme Park   

                                             Message Highest_Sentiment  \
0  🎟️🚨 Today's the FINAL day to use your 2023 Sea...               neu   

     New_Date      Time  Number of Reactions  Number of Comments (primary)  \
0  2023-12-31  21:40:00                 16.0                           2.0   

   Number of Shares.1  Comments per post  Comments on posts by fans  \
0                 2.0                2.0                        2.0   

  Post interaction rate Engagement  Number of Likes  Link-posts  \
0              1,09E+12   1,09E+12             13.0         NaN   

   Picture Posts  Video-posts  \
0            1.0          NaN   

                                                Link  \
0  https://www.facebook.com/617429960430190/posts...   

                                 Sentiments  
0  {'neg': 0.0, 'neu': 0.687, 'pos': 0.313}

## 2. Sentiment Analysis

In [ ]:
# Understand the sentiment of feature 'Message' by using sentiment analysis
# Import library natural language toolkit
import nltk
from nltk.sentiment import SentimentIntensityAnalyzer

In [ ]:
# Download the VADER lexicon
nltk.download('vader_lexicon')

# Initialize the VADER sentiment intensity analyzer
sia = SentimentIntensityAnalyzer()

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\yunus\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [ ]:
# Example messages to analyze
messages = [
    "Today's the FINAL day to use your 2023 Sea...",
    "See you next year! 😊 Don't miss...",
    "Don't miss our HUGE firework display! 🎆",
    # Add all other messages here
]

# Function to analyze the sentiment of each message
def analyze_sentiment(messages):
    results = []
    for message in messages:
        # Get the sentiment scores for each message
        sentiment_score = sia.polarity_scores(message)
        results.append(sentiment_score)
    return results

# Call the function and print the results
sentiment_results = analyze_sentiment(messages)
for message, sentiment in zip(messages, sentiment_results):
    print(f"Message: {message}")
    print(f"Sentiment: {sentiment}\n")

Message: Today's the FINAL day to use your 2023 Sea...
Sentiment: {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}

Message: See you next year! 😊 Don't miss...
Sentiment: {'neg': 0.0, 'neu': 1.0, 'pos': 0.0, 'compound': 0.0}

Message: Don't miss our HUGE firework display! 🎆
Sentiment: {'neg': 0.339, 'neu': 0.485, 'pos': 0.175, 'compound': -0.3297}



In [424]:
# Make sure you have the VADER lexicon downloaded
nltk.download('vader_lexicon')

# Initialize the VADER sentiment intensity analyzer
sia = SentimentIntensityAnalyzer()

# Assuming 'df' is your dataframe with a 'Message' column
merged_themepark['Sentiments'] = merged_themepark['Message'].apply(lambda x: sia.polarity_scores(x))

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\yunus\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [425]:
# Create a dataframe
df_themepark = merged_themepark

# Define a function to extract the highest sentiment
def extract_highest_sentiment(sentiments):
    # Exclude the 'compound' score as it's a combined score, not a separate sentiment
    sentiments.pop('compound', None)
    # Return the key with the highest value
    return max(sentiments, key=sentiments.get)

# Apply the function to the 'Sentiments' column to create the 'Highest_Sentiment' column
df_themepark['Highest_Sentiment'] = df_themepark['Sentiments'].apply(extract_highest_sentiment)

# Insert the 'Highest_Sentiment' column right after the 'Message' column
message_col_index = df_themepark.columns.get_loc('Message')
df_themepark.insert(message_col_index + 1, 'Highest_Sentiment', df_themepark.pop('Highest_Sentiment'))

2.1 Create a new DataFrame that insert a new feature named: Highest_Sentiment

In [426]:
df_themepark.head(1)

Profile-ID                      Post-ID                     Profile  \
0  84679583132  84679583132_770120045161180  Wild Adventures Theme Park   

                                             Message Highest_Sentiment  \
0  🎟️🚨 Today's the FINAL day to use your 2023 Sea...               neu   

     New_Date      Time  Number of Reactions  Number of Comments (primary)  \
0  2023-12-31  21:40:00                 16.0                           2.0   

   Number of Shares.1  Comments per post  Comments on posts by fans  \
0                 2.0                2.0                        2.0   

  Post interaction rate Engagement  Number of Likes  Link-posts  \
0              1,09E+12   1,09E+12             13.0         NaN   

   Picture Posts  Video-posts  \
0            1.0          NaN   

                                                Link  \
0  https://www.facebook.com/617429960430190/posts...   

                                 Sentiments  
0  {'neg': 0.0, 'neu': 0.687, 'pos': 0.313}

In [ ]:
sentiment_counts = df_themepark['Highest_Sentiment'].value_counts()

In [ ]:
sentiment_counts

Highest_Sentiment
neu    14331
neg      863
pos      249
Name: count, dtype: int64

The DataDrame sentiment_counts has the most neutral messages. What we found interesting is that negative messages be 2th. Therefore, we tend to analyze our sentiments and other features.

In [ ]:
# Splitting the dataframe
df_neg = df_themepark[df_themepark['Highest_Sentiment'] == 'neg']
df_neu = df_themepark[df_themepark['Highest_Sentiment'] == 'neu']
df_pos = df_themepark[df_themepark['Highest_Sentiment'] == 'pos']

In [ ]:
df_neu.describe()

Number of Reactions  Number of Comments (primary)  Number of Shares.1  \
count         14331.000000                  14331.000000        14331.000000   
mean            487.070756                     34.174517           62.403391   
std            3817.262891                    186.160289          521.417434   
min               0.000000                      0.000000            0.000000   
25%              23.000000                      1.000000            2.000000   
50%              68.000000                      6.000000            6.000000   
75%             215.000000                     18.000000           22.000000   
max          210756.000000                   8379.000000        30762.000000   

       Comments per post  Comments on posts by fans  Number of Likes  \
count       14331.000000               14331.000000     14331.000000   
mean           52.264741                  52.264741       349.879980   
std           268.604545                 268.604545      2574.086578   
min             0.000000                   0.000000         0.000000   
25%             1.000000                   1.000000        19.500000   
50%             8.000000                   8.000000        56.000000   
75%            27.000000                  27.000000       176.000000   
max         11170.000000               11170.000000    124591.000000   

       Link-posts  Picture Posts  Video-posts  
count       510.0        10667.0       2358.0  
mean          1.0            1.0          1.0  
std           0.0            0.0          0.0  
min           1.0            1.0          1.0  
25%           1.0            1.0          1.0  
50%           1.0            1.0          1.0  
75%           1.0            1.0          1.0  
max           1.0            1.0          1.0

- Records the engagement metrics for approximately 14,331 entries
- On average, there are about 487 reactions per entry with a wide variance
- Comments seem less frequent, averaging around 34 per entry, but with some entries reaching into the thousand
- The mean number of shares is higher than comments, suggesting shares might be a more common form of engagement in this dataset
- Comments per post, likes, and other types of posts have a broad range of values but generally follow the trends of reactions and shares

In [ ]:
df_neg.describe()

Number of Reactions  Number of Comments (primary)  Number of Shares.1  \
count           863.000000                    863.000000          863.000000   
mean            557.965238                     36.792584           46.438007   
std            1978.230677                    137.538343          183.290453   
min               0.000000                      0.000000            0.000000   
25%              12.000000                      0.000000            0.000000   
50%              38.000000                      2.000000            3.000000   
75%             226.000000                     13.000000           15.000000   
max           26908.000000                   1707.000000         2102.000000   

       Comments per post  Comments on posts by fans  Number of Likes  \
count         863.000000                 863.000000       863.000000   
mean           67.844728                  67.844728       411.891078   
std           254.223181                 254.223181      1449.451172   
min             0.000000                   0.000000         0.000000   
25%             0.000000                   0.000000        10.000000   
50%             3.000000                   3.000000        32.000000   
75%            19.000000                  19.000000       183.000000   
max          2908.000000                2908.000000     21444.000000   

       Link-posts  Picture Posts  Video-posts  
count        39.0          720.0         74.0  
mean          1.0            1.0          1.0  
std           0.0            0.0          0.0  
min           1.0            1.0          1.0  
25%           1.0            1.0          1.0  
50%           1.0            1.0          1.0  
75%           1.0            1.0          1.0  
max           1.0            1.0          1.0

- Consists of 863 entries with engagement metrics
- Entries receive a higher average number of reactions than the first dataset, around 558, but the total count is much lower
- The distribution of reactions and comments suggests more variance with a higher average but lower total count compared to the first dataset
- This dataset also measures comments per post, comments by fans, likes, and the number of specific post types (link, picture, video), with each category having its own range and distribution


In [ ]:
df_pos.describe()

Number of Reactions  Number of Comments (primary)  Number of Shares.1  \
count           249.000000                    249.000000          249.000000   
mean            688.658635                     26.662651           97.574297   
std            2162.006160                     58.953428          485.962575   
min               0.000000                      0.000000            0.000000   
25%              25.000000                      1.000000            1.000000   
50%              83.000000                      6.000000            5.000000   
75%             412.000000                     23.000000           29.000000   
max           22625.000000                    417.000000         6314.000000   

       Comments per post  Comments on posts by fans  Number of Likes  \
count         249.000000                 249.000000       249.000000   
mean           41.060241                  41.060241       499.851406   
std            91.346869                  91.346869      1437.272363   
min             0.000000                   0.000000         0.000000   
25%             2.000000                   2.000000        22.000000   
50%             7.000000                   7.000000        68.000000   
75%            37.000000                  37.000000       322.000000   
max           671.000000                 671.000000     12182.000000   

       Link-posts  Picture Posts  Video-posts  
count         6.0          175.0         47.0  
mean          1.0            1.0          1.0  
std           0.0            0.0          0.0  
min           1.0            1.0          1.0  
25%           1.0            1.0          1.0  
50%           1.0            1.0          1.0  
75%           1.0            1.0          1.0  
max           1.0            1.0          1.0

- Has a total of 249 entries, which is smaller than the other two datasets
- The average reactions per entry are the highest among the three datasets at about 689, but with considerable variability
- The number of comments and shares per entry also shows a high mean value, indicating a possibly engaged audience
- Similar to the other datasets, there is a count of different post types, with likes again showing a high mean, suggesting that 'likes' are a common interaction
